app/state_machines/form_state_machine.rb

```diff
@@ -0,0 +1,60 @@
+module FormStateMachine
+  extend ActiveSupport::Concern
+
+  included do
+    include AASM
+
+    enum :state, {
+      draft: "draft",
+      live: "live",
+      draft_live: "draft_live",
+      archived: "archived",
+      draft_archived: "draft_archived",
+    }
+
+    aasm column: :state, enum: true do
+      state :draft, initial: true
+      state :live, :draft_live, :archived, :draft_archived
```

Comment:
Am I right in thinking these are roughly what the different states actually are? 
draft: Has no live snapshot
live: Matches most recent live snapshot
draft_live: Doesn't match most recent live snapshot
archived: Archived and matches live snapshot
draft_archived: Archived and doesn't match live snapshot

Reply:
these are the values that would be stored in the `state` column of the forms table. A form would only ever have 1 state at a time. its not tied to any of the snapshots but you could still use `form.live_version` to get the snapshot (if it exists) in any states where you mentioned `snapshot` above

Reply:
Yeah that makes sense - was just checking whether I roughly understand what those states are meant to mean to us